src/analytics/reports/periodReport/cohortRetentionReport.ts

```diff
@@ -131,3 +130,204 @@ function createUniqueUsersCohort(
     }),
   ];
 }
+
+/**
+ * @param cohort [num_users_at_start, num_users_after_1_period, ...]
+ * @returns [num_users_at_start, num_and_perc_users_after_1_period, ...]
+ *
+ * Examples of returned value:
+ *    - `["10", "6 (60%)", "3 (30%)", "0 (0%)"]`
+ *    - `["0", "N/A", "N/A"]`
+ */
+function calcCohortRetentionTableRow(cohort: number[]): string[] {
+  const [numUsersAtStart, ...numUsersThroughPeriods] = cohort;
+  const numUsersWithRetentionPercentagesThroughPeriods =
+    numUsersThroughPeriods.map((n) =>
+      numUsersAtStart === 0
+        ? "N/A"
+        : `${n} (${Math.round((n / numUsersAtStart) * 100)}%)`,
+    );
+  return [
+    numUsersAtStart.toString(),
+    ...numUsersWithRetentionPercentagesThroughPeriods,
+  ];
+}
+
+type CohortRetentionHeatMapPoint = MatrixDataPoint & {
+  size: number;
+  retentionPercentage: number;
+};
+
+async function createCohortRetentionHeatMap(
+  cohorts: number[][],
+  periods: Period[],
+  periodName: PeriodName,
+): Promise<Buffer> {
+  const initialPeriodColorInterpolator = createColorInterpolator(
+    SEQUENTIAL_GREEN_PALETTE,
+  );
+  const afterInitialPeriodColorInterpolator = createColorInterpolator(
+    SEQUENTIAL_BLUE_PALETTE,
+  );
+
+  const maxSize = Math.max(...cohorts.map((cohort) => cohort[0]));
+  const maxCohortRetentionPercentageAfterInitialPeriod: number = Math.max(
+    ...cohorts.flatMap((cohort) =>
+      cohort.map((value) => value / cohort[0]).slice(1),
```

Comment:
Makes a bit more sense:
```suggestion
      cohort.slice(1).map((value) => value / cohort[0]),
```